jax/_src/pallas/pallas_call.py

```diff
@@ -1878,8 +1878,10 @@ def wrapped(*args):
             f"a different abstract value {out_aval}.")
 
     index_args, rest_args = split_list(flat_args, [grid_mapping.num_index_operands])
-    with pallas_core.interpret_mode_env(interpret):
-      out_flat = pallas_call_p.bind(
+    if interpret and backend == 'mosaic_tpu':
```

Comment:
Can we just add this as an alternative backend to pallas_call and register it in the MLIR lowering? I think it's confusing to have multiple interpret modes (interpret itself isn't even a great name). You would probably just have to call MLIR's lower_fun on your interpreter.